testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveBasicTableStatistics.java

```diff
@@ -199,25 +199,38 @@ public void testAnalyzePartitioned(boolean transactional)
-            if (tableStatistics.getNumRows().isEmpty()) {
-                assertThatStatisticsAreNotPresent(tableStatistics);
+            if (tableStatisticsBefore.getNumRows().isEmpty()) {
+                assertThatStatisticsAreNotPresent(tableStatisticsBefore);
             }
 
             BasicStatistics partitionStatisticsBefore = getBasicStatisticsForPartition(onHive(), tableName, "n_regionkey=1");
             assertThatStatisticsArePresent(partitionStatisticsBefore);
 
             // run ANALYZE
             onTrino().executeQuery(format("ANALYZE %s", tableName));
-            BasicStatistics partitionStatisticsAfter = getBasicStatisticsForPartition(onHive(), tableName, "n_regionkey=1");
-            assertThatStatisticsArePresent(partitionStatisticsAfter);
+            BasicStatistics partitionStatisticsAfterAnalyze = getBasicStatisticsForPartition(onHive(), tableName, "n_regionkey=1");
+            assertThatStatisticsArePresent(partitionStatisticsAfterAnalyze);
 
             // ANALYZE must not change the basic stats
-            assertThat(partitionStatisticsBefore.getNumRows().getAsLong()).isEqualTo(partitionStatisticsAfter.getNumRows().getAsLong());
-            assertThat(partitionStatisticsBefore.getNumFiles().getAsLong()).isEqualTo(partitionStatisticsAfter.getNumFiles().getAsLong());
-            assertThat(partitionStatisticsBefore.getRawDataSize().getAsLong()).isEqualTo(partitionStatisticsAfter.getRawDataSize().getAsLong());
-            assertThat(partitionStatisticsBefore.getTotalSize().getAsLong()).isEqualTo(partitionStatisticsAfter.getTotalSize().getAsLong());
+            assertThat(partitionStatisticsBefore.getNumRows().getAsLong()).isEqualTo(partitionStatisticsAfterAnalyze.getNumRows().getAsLong());
+            assertThat(partitionStatisticsBefore.getNumFiles().getAsLong()).isEqualTo(partitionStatisticsAfterAnalyze.getNumFiles().getAsLong());
+            assertThat(partitionStatisticsBefore.getRawDataSize().getAsLong()).isEqualTo(partitionStatisticsAfterAnalyze.getRawDataSize().getAsLong());
+            assertThat(partitionStatisticsBefore.getTotalSize().getAsLong()).isEqualTo(partitionStatisticsAfterAnalyze.getTotalSize().getAsLong());
+
+            // Insert more rows
+            onTrino().executeQuery(format("INSERT INTO %s (n_nationkey, n_name, n_comment, n_regionkey) VALUES (12, 'Utopia', 'Fun here', 1), (13, 'Perdition', 'Getting hot', 1)", tableName));
+            BasicStatistics partitionStatisticsAfterInsert = getBasicStatisticsForPartition(onHive(), tableName, "n_regionkey=1");
+
+            // Check that statistics were updated
+            assertThat(partitionStatisticsBefore.getNumRows().getAsLong() + 2).isEqualTo(partitionStatisticsAfterInsert.getNumRows().getAsLong());
+            assertThat(partitionStatisticsBefore.getNumFiles().getAsLong() + 2).isEqualTo(partitionStatisticsAfterInsert.getNumFiles().getAsLong());
```

Comment:
nit: it is `assertThat(value).isEqualTo(expected)` so it would be more proper to write:

```java
assertThat(partitionStatisticsAfterInsert.getNumRows().getAsLong()).isEqualTo( partitionStatisticsBefore.getNumRows().getAsLong() + 2);
```